<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RandomNumberClass</class>
 <widget class="QWidget" name="RandomNumberClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RandomNumber</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <layout class="QVBoxLayout" name="MainVerticalLayout">
     <item>
      <layout class="QHBoxLayout" name="MainLayout">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QComboBox" name="NumberTypeBox"/>
         </item>
         <item>
          <widget class="QWidget" name="IntegerWidget" native="true">
           <layout class="QHBoxLayout" name="IntegerLayout">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Min:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="MinIntBox">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Orientation::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Max:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="MaxIntBox">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
              <property name="value">
               <number>10</number>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="DoubleWidget" native="true">
           <layout class="QHBoxLayout" name="DoubleLayout">
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Min:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="MinDoubleBox">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Orientation::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Max:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="MaxDoubleBox">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
              <property name="value">
               <double>1.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="OutputField">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="NumbersLayout">
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Numbers:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="NumberCountBox">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1000000</number>
         </property>
         <property name="value">
          <number>1</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="GenerateButton">
       <property name="text">
        <string>Generate</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>GenerateButton</sender>
   <signal>clicked()</signal>
   <receiver>RandomNumberClass</receiver>
   <slot>OnGenerateButtonPress()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>377</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NumberTypeBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>RandomNumberClass</receiver>
   <slot>OnNumberTypeBoxChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MaxDoubleBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>RandomNumberClass</receiver>
   <slot>OnSpinBoxChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>275</x>
     <y>256</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MaxIntBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>RandomNumberClass</receiver>
   <slot>OnSpinBoxChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>275</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MinDoubleBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>RandomNumberClass</receiver>
   <slot>OnSpinBoxChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>256</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MinIntBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>RandomNumberClass</receiver>
   <slot>OnSpinBoxChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OnGenerateButtonPress()</slot>
  <slot>OnNumberTypeBoxChange(int)</slot>
  <slot>OnSpinBoxChange()</slot>
 </slots>
</ui>
